lib/spack/spack/environment.py

```diff
@@ -1507,13 +1516,14 @@ def write(self, regenerate_views=True):
                         del yaml_dict[key]
 
         # if all that worked, write out the manifest file at the top level
-        # Only actually write if it has changed or was never written
-        changed = self.yaml != self.raw_yaml
-        written = os.path.exists(self.manifest_path)
-        if changed or not written:
-            self.raw_yaml = copy.deepcopy(self.yaml)
-            with fs.write_tmp_and_move(self.manifest_path) as f:
-                _write_yaml(self.yaml, f)
+        # (we used to check whether the yaml had changed and not write it out
+        # if it hadn't. We can't do that anymore because it could be the only
+        # thing that changed is the "override" attribute on a config dict,
+        # which would not show up in even a string comparison between the two
+        # keys).
+        self.raw_yaml = copy.deepcopy(self.yaml)
+        with fs.write_tmp_and_move(self.manifest_path) as f:
+            _write_yaml(self.yaml, f)
```

Comment:
I think we need to talk about how to get around this -- if the override attribute changes, we should be able to see that somehow and avoid writing out when there are no changes.  I *believe* there were some prior issues/annoyances with this -- not least of which is that the `spack.yaml` would change just a little bit in repos where it hadn't been modified.  That would annoy developers.

Reply:
I got around it by writing my own comparator locally in `spack.environment`.